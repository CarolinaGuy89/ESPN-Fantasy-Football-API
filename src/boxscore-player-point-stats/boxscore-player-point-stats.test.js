import ApiModel from '../api-model/api-model.js';
import BoxscorePlayerPointStats from './boxscore-player-point-stats.js';

import { localObject, serverResponse } from './boxscore-player-point-stats.stubs.js';

describe('BoxscorePlayerPointStats', () => {
  let boxscorePlayerPointStats;

  beforeEach(() => {
    boxscorePlayerPointStats = new BoxscorePlayerPointStats();
  });

  afterEach(() => {
    boxscorePlayerPointStats = null;
  });

  test('extends ApiModel', () => {
    expect(boxscorePlayerPointStats).toBeInstanceOf(ApiModel);
  });

  describe('attribute population from server response', () => {
    beforeEach(() => {
      boxscorePlayerPointStats = BoxscorePlayerPointStats.buildFromServer(serverResponse);
    });

    test('parses data correctly', () => {
      expect(boxscorePlayerPointStats).toMatchSnapshot();
    });
  });

  describe('attribute population from local object', () => {
    beforeEach(() => {
      boxscorePlayerPointStats = BoxscorePlayerPointStats.buildFromLocal(localObject);
    });

    test('parses data correctly', () => {
      expect(boxscorePlayerPointStats).toMatchSnapshot();
    });
  });

  describe('class methods', () => {
    test('throws error', () => {
      expect(() => BoxscorePlayerPointStats.read()).toThrowError(
        `${BoxscorePlayerPointStats.displayName}: read: Cannot call read.`
      );
    });
  });
});
