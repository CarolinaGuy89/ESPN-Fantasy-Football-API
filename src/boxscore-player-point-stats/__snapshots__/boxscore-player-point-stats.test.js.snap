// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxscorePlayerPointStats attribute population from local object parses data correctly 1`] = `
BoxscorePlayerPointStats {
  "defensive0PointsAllowed": 10,
  "defensive100To199YardsAllowed": -10,
  "defensive14To17PointsAllowed": 10,
  "defensive1To6PointsAllowed": 10,
  "defensive200To299YardsAllowed": 10,
  "defensive28To34PointsAllowed": 10,
  "defensive350To399YardsAllowed": -10,
  "defensive35To45PointsAllowed": 10,
  "defensive400To449YardsAllowed": 10,
  "defensive450To499YardsAllowed": 10,
  "defensive500To549YardsAllowed": 10,
  "defensive7To13PointsAllowed": 10,
  "defensiveBlockedKickForTouchdowns": 10,
  "defensiveBlockedKicks": 10,
  "defensiveFumbles": 10,
  "defensiveInterceptions": 10,
  "defensiveOver550YardsAllowed": 10,
  "defensiveSacks": 10,
  "defensiveSafeties": 10,
  "fumbleReturnTouchdown": 10,
  "interceptionReturnTouchdown": 10,
  "kickoffReturnTouchdown": 10,
  "lostFumbles": 10,
  "madeExtraPoints": 10,
  "madeFieldGoalsFrom40To49": 10,
  "madeFieldGoalsFrom50Plus": 10,
  "madeFieldGoalsFromUnder40": 10,
  "missedExtraPoints": 10,
  "missedFieldGoals": 10,
  "passing2PtConversions": 10,
  "passingInterceptions": 10,
  "passingTouchdowns": 10,
  "passingYards": 10,
  "puntReturnTouchdown": 10,
  "receiving2PtConversions": 10,
  "receivingReceptions": 10,
  "receivingTouchdowns": 10,
  "receivingYards": 10,
  "rushing2PtConversions": 10,
  "rushingTouchdowns": 10,
  "rushingYards": 10,
}
`;

exports[`BoxscorePlayerPointStats attribute population from server response parses data correctly 1`] = `
BoxscorePlayerPointStats {
  "defensive0PointsAllowed": 10,
  "defensive100To199YardsAllowed": -10,
  "defensive14To17PointsAllowed": 10,
  "defensive1To6PointsAllowed": 10,
  "defensive200To299YardsAllowed": 10,
  "defensive28To34PointsAllowed": 10,
  "defensive350To399YardsAllowed": -10,
  "defensive35To45PointsAllowed": 10,
  "defensive400To449YardsAllowed": 10,
  "defensive450To499YardsAllowed": 10,
  "defensive500To549YardsAllowed": 10,
  "defensive7To13PointsAllowed": 10,
  "defensiveBlockedKickForTouchdowns": 10,
  "defensiveBlockedKicks": 10,
  "defensiveFumbles": 10,
  "defensiveInterceptions": 10,
  "defensiveOver550YardsAllowed": 10,
  "defensiveSacks": 10,
  "defensiveSafeties": 10,
  "fumbleReturnTouchdown": 10,
  "interceptionReturnTouchdown": 10,
  "kickoffReturnTouchdown": 10,
  "lostFumbles": 10,
  "madeExtraPoints": 10,
  "madeFieldGoalsFrom40To49": 10,
  "madeFieldGoalsFrom50Plus": 10,
  "madeFieldGoalsFromUnder40": 10,
  "missedExtraPoints": 10,
  "missedFieldGoals": 10,
  "passing2PtConversions": 10,
  "passingInterceptions": 10,
  "passingTouchdowns": 10,
  "passingYards": 10,
  "puntReturnTouchdown": 10,
  "receiving2PtConversions": 10,
  "receivingReceptions": 10,
  "receivingTouchdowns": 10,
  "receivingYards": 10,
  "rushing2PtConversions": 10,
  "rushingTouchdowns": 10,
  "rushingYards": 10,
}
`;
